### ISSUE TITLE
[Slider] Distinguish onInput from onValueChange

### ISSUE THREAD

**Issue by asherccohen** [radix-ui/primitives#903]
## Feature request
Slider should have two triggers.
One to act onInput and one to act on blur that would fire onValueChange.

See discussion at the bottom of this issue:

https://github.com/radix-ui/primitives/issues/570

### Overview

Like in the demo created by @jjenzz it is typical for sliders (as well as inputs) to denounce the value so that devs can render the real-time value but only trigger an update/network request onFinalChange.

The problem with the demo is that the denounced value would render with a delay(obviously), while the use case is to render real-time values in the UI and send an update (redux/remote/etc) once dragging is finished.

I understand radix provides minimal API so I'm not asking for the value to be denounced, just the events to be separated into two distinct actions.

### Examples in other libraries

An example can also be found on the very popular react-range repo.
https://github.com/tajo/react-range

### Who does this impact? Who is this for?

This is for all users as it's a very common use case.

### Additional context

Thank you!

**Comment by jjenzz** [comment]
> Additionally, when using the keyboard then is the commit on every change?

For each component we decided to support these two events for, we could try to match the native behaviour as closely as possible. In the case of `Slider`, you're correct that it would be every change for keyboard and pointer up for pointer users:

https://codepen.io/jjenzz/pen/PoKwPYP

**Comment by benoitgrelard** [comment]
Ah that's cool, I didn't even think to check the native one.
I didn't really answer your question though. I can't see ourselves having to rename all the onChange handlers because of that, so the suggestion of a `onCommit` sort of make sense? But then would that make us inconcistent with React itself?

**Comment by benoitgrelard** [comment]
Oh I see, I read back and saw you meant relative to the native `oninput` and `onchange` events.
I think my vote would go to add additional `onValueCommit` events in components where it makes sense (ie. components that don't have just discrete changes).

**Comment by lucat1** [comment]
> In the meantime, it can be controlled to achieve the expectation https://codesandbox.io/s/dazzling-mahavira-u7yr6?file=/App.js

I'd like to point out that the provided example does not behave properly: When I interact witth the slider without using the thumb the `pointerup` event is not fired as it's only being handled on the `StyledThumb` element. A quick fix can be achieved by applying the listener on the `Root` element and have the event propagated from both the thumb and the track. An alternative is to duplicate the handler on both elements.

**Comment by JClackett** [comment]
Just wanted to check something here, when I use the keyboard with the slider and use onValueChange and onValueCommit, I noticed that the onValueCommit runs before the onValueChange, is this correct?

I have some logic that requires the commit callback to run after the change

**Comment by benoitgrelard** [comment]
Hey @JClackett, that's correct but it is merely because of the implementation. Right now the way they are tied together would make it very difficult to change that.

However, it largely shouldn't be an issue as they are happening in the same tick/event loop.

> I have some logic that requires the commit callback to run after the change

Can you expand on this as I don't think the order should be relied on. Also, even if they were reversed it happens so in the same tick, so I'm not sure that would give enough time to do anything anyway.

**Comment by JClackett** [comment]
@benoitgrelard so my use case is that im using the slider as a sort of music player scrub bar. When sliding, I set a state of "SEEKING" (using a state machine for all of this) and update the slider value so the slider moves around, and on finish (i.e. onValueCommit), I actually set the players time to the value of the slider. I need to set this SEEKING state on slide as the player itself sets the current time whilst the song plays so the slider moves on its own (when the user is manually scrubbing we need disable this, so it doesn't jump all over the place).

I was using a regular input of type="range" before with:

```
onMouseDown
onTouchStart
onKeyDown
```
to set SEEKING, and:

```
onMouseUp
onTouchEnd
onKeyUp
```

to set FINISHED_SEEKING


But styling the input range is annoying!
